pulsar/consumer_impl.go

```diff
@@ -74,6 +74,11 @@ func newConsumer(client *client, options ConsumerOptions) (Consumer, error) {
 		options.ReceiverQueueSize = 1000
 	}
 
+	// disable receiver queue if queue size is negative
+	if options.ReceiverQueueSize < 0 {
+		options.ReceiverQueueSize = 0
+	}
```

Comment:
Maybe we can fix it using follows way:

```
	if options.ReceiverQueueSize <= 0 {
		options.ReceiverQueueSize = 1000
	}
```

Reply:
unbuffered chan may be useful for some cases.

Reply:
not sure whether it's a good idea to remove this option. If negative value is totally disallowed (should be reflected in comments), i think it's ok to panic though.

Reply:
This is related to the flow mechanism of messages in pulsar. By default, the value of the receiveQueueSize is 1000

Reply:
Ok, i'll update the modification and comments according to your suggestions.